File: Day7/lesson.py
# Integrated example: Class, functions, file I/O, error handling.

class User:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def save(self, filename):
        try:
            with open(filename, "w") as f:
                f.write(f"{self.name} {self.age}\n")
        except IOError as e:
            print(f"Save failed: {e}")

    @classmethod
    def load(cls, filename):
        try:
            with open(filename, "r") as f:
                data = f.read().strip().split()
                return cls(data[0], int(data[1]))
        except (IOError, ValueError) as e:
            print(f"Load failed: {e}")
            return cls("Error", -1)

# Usage
u = User("Bob", 25)
u.save("user.txt")
loaded = User.load("user.txt")
print(f"Loaded: {loaded.name}, {loaded.age}")

File: Day7/lesson.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Integrated example: Struct, functions, file I/O, error handling.

struct User {
    char name[50];
    int age;
};

void save_user(struct User u) {
    FILE *f = fopen("user.txt", "w");
    if (f == NULL) {
        perror("Save failed");
        return;
    }
    fprintf(f, "%s %d\n", u.name, u.age);
    fclose(f);
}

struct User load_user() {
    struct User u;
    FILE *f = fopen("user.txt", "r");
    if (f == NULL) {
        perror("Load failed");
        strcpy(u.name, "Error");
        u.age = -1;
        return u;
    }
    fscanf(f, "%s %d", u.name, &u.age);
    fclose(f);
    return u;
}

int main() {
    struct User u = {"Bob", 25};
    save_user(u);
    struct User loaded = load_user();
    printf("Loaded: %s, %d\n", loaded.name, loaded.age);
    return 0;
}
